eth/client_ticketbroker.go

```diff
@@ -62,17 +67,52 @@ func (c *client) RedeemWinningTicket(ticket *pm.Ticket, sig []byte, recipientRan
 
 // GetSenderInfo returns the info for a sender
 func (c *client) GetSenderInfo(addr ethcommon.Address) (*pm.SenderInfo, error) {
-	info, err := c.TicketBrokerSession.GetSenderInfo(addr)
+	info := new(struct {
```

Comment:
We might be able to avoid this custom low level call altogether by generating the contract bindings using a version of `abigen` built from this branch https://github.com/ethereum/go-ethereum/pull/20179 which will use the `internalType` specified in a contract's ABI to generate Go structs with meaningful names (another benefit is that we won't end up with struct names like `Struct0` and `Struct1` in the contract binding).

Reply:
I'm hesitant to include any generated code by that branch in our own codebase for now, seeing as CI checks for that branch do not pass and is likely due for change requests. 

While it is surely a step in the right direction I think it might be good to be a little conservative here and address it in a seperate PR when that branch becomes merged. WDYT?

Reply:
Agree that we can update the bindings in a separate PR.

I think we can slim down this implementation a bit by instantiating a `bind.BoundContract` type using the contract ABI. Here is a sketch (have not tested this):

```
// GetSenderInfo returns the info for a sender
func (c *client) GetSenderInfo(addr ethcommon.Address) (*pm.SenderInfo, error) {
	out := new(struct {
		Sender struct {
			Deposit       *big.Int
			WithdrawRound *big.Int
		}
		Reserve pm.ReserveInfo
	})

	abi, err := abi.JSON(strings.NewReader(contracts.TicketBrokerABI))
	if err != nil {
		return nil, err
	}

	contract := bind.NewBoundContract(c.ticketBrokerAddr, abi, c.backend, c.backend, c.backend)
	if err := contract.Call(&c.TicketBrokerSession.CallOpts, out, "getSenderInfo", addr); err != nil {
		return nil, err
	}

	return &pm.SenderInfo{
		Deposit:       out.Sender.Deposit,
		WithdrawRound: out.Sender.WithdrawRound,
		Reserve: &pm.ReserveInfo{
			FundsRemaining:        out.Reserve.FundsRemaining,
			ClaimedInCurrentRound: out.Reserve.ClaimedInCurrentRound,
		},
	}, nil
}
```

Reply:
fixed in 583a3b5

I forgot to drop go modules changes in the commit after updating to new LPMS version but I'll get those in the rebase. 

That being said I didn't run into any issues running this in the test-harness so should be g2g. 

The block header not found issue seems to be occuring on current `master` +  `pm` as well so something we can track in a separate issue (I'll create one?).

`I1021 18:52:22.967576       1 block_watcher.go:167] block header not found blockNumber=946`